fdbserver/DataDistribution.actor.cpp

```diff
@@ -3713,9 +3716,106 @@ static std::set<int> const& normalDataDistributorErrors() {
 	return s;
 }
 
+ACTOR Future<Void> ddSnapCreate(DistributorSnapRequest snapReq, Reference<AsyncVar<struct ServerDBInfo>> db ) {
+	state Database cx = openDBOnServer(db, TaskDefaultEndpoint, true, true);
+	state double snapTimeout = g_network->isSimulated() ? 10.0 : SERVER_KNOBS->SNAP_CREATE_MAX_TIMEOUT;
+	TraceEvent("SnapDataDistributor.SnapReqEnter")
+		.detail("SnapPayload", snapReq.snapPayload)
+		.detail("SnapUID", snapReq.snapUID);
+	try {
+		// disable tlog pop on local tlog nodes
+		std::vector<TLogInterface> tlogs = db->get().logSystemConfig.allLocalLogs();
+		state std::vector<Future<Void>> disablePops;
+		for (const auto & tlog : tlogs) {
+			disablePops.push_back(
+				timeoutError(tlog.disablePopRequest.tryGetReply(TLogDisablePopRequest(snapReq.snapUID)), snapTimeout)
+				);
+		}
+		wait(waitForAll(disablePops));
+
+		TraceEvent("SnapDataDistributor.AfterDisableTLogPop")
+			.detail("SnapPayload", snapReq.snapPayload)
+			.detail("SnapUID", snapReq.snapUID);
+		// snap local storage nodes
+		std::vector<WorkerInterface> storageWorkers = wait(timeoutError(getStorageWorkers(cx, db, true /* localOnly */), snapTimeout));
```

Comment:
you are still passing the snapTimeout around everywhere. I would prefer to wrap the whole actor within a timeout instead.